packages/core/src/components/button/buttons.tsx

```diff
@@ -73,13 +87,27 @@ export const AnchorButtonFactory = React.createFactory(AnchorButton);
 function getButtonClasses(props: IButtonProps) {
     return classNames(
         Classes.BUTTON,
-        { [Classes.DISABLED]: props.disabled },
+        { [Classes.DISABLED]: isButtonDisabled(props) },
         Classes.iconClass(props.iconName),
         Classes.intentClass(props.intent),
         props.className,
     );
 }
 
+function isButtonDisabled(props: IButtonProps) {
+    return props.disabled === true || props.loading === true;
+}
+
+function buttonContentsVisibility(loading: boolean) {
+    return {visibility: loading ? "hidden" : "visible"};
```

Comment:
i see why this is tricky to work around and the `span` may be necessary, but it's the one thing in this PR that makes me most uncomfortable.